ste/jobStatusManager.go

```diff
@@ -78,41 +137,38 @@ func (jm *jobMgr) handleStatusUpdateMessage() {
 
 	for {
 		select {
-		case msg := <-jstm.partCreated:
-			js.CompleteJobOrdered = js.CompleteJobOrdered || msg.IsFinalPart
-			js.TotalTransfers += msg.TotalTransfers
-			js.FileTransfers += msg.FileTransfers
-			js.FolderPropertyTransfers += msg.FolderTransfer
-			js.TotalBytesEnumerated += msg.TotalBytesEnumerated
-			js.TotalBytesExpected += msg.TotalBytesEnumerated
-
-		case msg := <-jstm.xferDone:
-			msg.Src = common.URLStringExtension(msg.Src).RedactSecretQueryParamForLogging()
-			msg.Dst = common.URLStringExtension(msg.Dst).RedactSecretQueryParamForLogging()
-
-			switch msg.TransferStatus {
-			case common.ETransferStatus.Success():
-				js.TransfersCompleted++
-				js.TotalBytesTransferred += msg.TransferSize
-			case common.ETransferStatus.Failed(),
-				common.ETransferStatus.TierAvailabilityCheckFailure(),
-				common.ETransferStatus.BlobTierFailure():
-				js.TransfersFailed++
-				js.FailedTransfers = append(js.FailedTransfers, msg)
-			case common.ETransferStatus.SkippedEntityAlreadyExists(),
-				common.ETransferStatus.SkippedBlobHasSnapshots():
-				js.TransfersSkipped++
-				js.SkippedTransfers = append(js.SkippedTransfers, msg)
-			}
+		// drainXferDoneSignal is high priority channel,process any message pending on xferDone channel.
+		case <-jstm.drainXferDoneSignal:
+			drainXferDoneCh(jm)
+
+		default:
+			select {
+			case msg := <-jstm.partCreated:
+				js.CompleteJobOrdered = js.CompleteJobOrdered || msg.IsFinalPart
+				js.TotalTransfers += msg.TotalTransfers
+				js.FileTransfers += msg.FileTransfers
+				js.FolderPropertyTransfers += msg.FolderTransfer
+				js.TotalBytesEnumerated += msg.TotalBytesEnumerated
+				js.TotalBytesExpected += msg.TotalBytesEnumerated
+
+			case msg := <-jstm.xferDone:
+				processXferDoneMsg(msg, js)
+
+			case <-jstm.listReq:
+				/* Display stats */
+				js.Timestamp = time.Now().UTC()
+				jstm.respChan <- *js
+
+			case <-jstm.drainXferDoneSignal:
+				drainXferDoneCh(jm)
+
+			case <-jstm.done:
+				fmt.Println("Cleanup JobStatusmgr")
+				jstm.doneHandleStatusUpdateManager = true
```

Comment:
nit, instead of doing this in all exit paths, better to do the following at L138

// Set jstm.doneHandleStatusUpdateManager when handleStatusUpdateMessage goroutine exits.
defer func() {
    jstm.doneHandleStatusUpdateManager = true
}